query-specifications/interactive-complex-read-14.yaml

```diff
@@ -3,18 +3,12 @@ operation: complex
 number: 14
 title: Trusted connection paths
 description: |
-  Given two *Persons*, find all (unweighted) shortest paths between these two *Persons*, in the subgraph induced by the *knows* relationship.
-  
-  Then, for each path calculate a weight. The nodes in the path are *Persons*, and the weight of a path is the sum of weights between every pair of consecutive *Person* nodes in the path.
-
-  The weight for a pair of *Persons* is calculated based on their interactions:
-
-  * Every direct reply (by one of the *Persons*) to a *Post* (by the other *Person*) contributes 1.0.
-  * Every direct reply (by one of the *Persons*) to a *Comment* (by the other *Person*) contributes 0.5.
+  Given two *Persons*, find a cheapest path between two *Persons*, in the interaction subgraph.
 
-  Note that interactions are counted both ways (e.g. if Alice writes 2 *Post* replies and 1 *Comment* reply to Bob, while Bob writes 3 *Post* replies and 4 *Comment* replies to Alice, their interaction score is $2 \times 1.0 + 1 \times 0.5 + 3 \times 1.0 + 4 \times 0.5 = 7.5$).
-
-  Return all the paths with shortest length, and their weights. Do not return any rows if there is no path between the two *Persons*.
+  The interaction subgraph is based on the Person-knows-Person graph where the Person endpoints have at least one interaction between them.
```

Comment:
Does `endpoints` refer to the supplied persons? So, if we have a shortest path of P1 -> P3 -> P2 would the criteria be that P2 has at least one interaction with P1? 

This would seem more real-world as it gives us some insight into community in the subraph of friends replying to friends of friends messages but just looking for clarification.

Reply:
No, "endpoints" are interpreted on each PKP edge. So P1-P2 do not have to be connected directly.